common/app/routes/challenges/components/Classic-Modal.jsx

```diff
@@ -0,0 +1,74 @@
+import React, { PropTypes } from 'react';
+import { Button, Modal } from 'react-bootstrap';
+import PureComponent from 'react-pure-render/component';
+import FontAwesome from 'react-fontawesome';
+
+export default class ClassicModal extends PureComponent {
+
+  keyBindings(event) {
+    const map = Object.assign({}, this.props.keyMap);
```

Comment:
```js
if (map[17] && map[13] && open) {
       submitChallenge();
     }
```

Without passing the key map into the store, I would only get `map[13]` or `map[17]` in each function call (keydown)